cairo/ethereum/cancun/fork_types.cairo

```diff
@@ -8,6 +8,10 @@ from ethereum.utils.numeric import is_zero
 
 using Address = Bytes20;
 
+struct OptionalAddress {
+    value: felt*,
```

Comment:
why is it a pointer to a felt and not an Address?

Reply:
because the way the Optional system was built in https://github.com/kkrt-labs/keth/pull/437 is that the optional type contains the inner type directly instead of being a pointer to the outer type, thus i'm just re-using the same principle

Reply:
and changing to a pointer to Address would require too many changes in the codebase to be worth (as it requires changing all usages of OptionalAccount as well)

Reply:
why is it a felt pointer and not a felt then?